crates/router/src/core/payment_methods/cards.rs

```diff
@@ -2113,6 +2119,18 @@ pub async fn filter_payment_methods(
     }
     Ok(())
 }
+fn filter_pm_based_on_supported_payments_for_update_mandate(
+    supported_payment_methods_for_mandate: &settings::SupportedPaymentMethodsForMandate,
+    payment_method_type: &api_enums::PaymentMethodType,
+    connector: api_enums::Connector,
+) -> bool {
+    supported_payment_methods_for_mandate
+        .0
+        .get(&api_enums::PaymentMethod::Card)
```

Comment:
Don't hardcode this to `Card`. We're using the Payment Method Type from the payment method object so the Payment Method should come from there too.

Reply:
So should we construct a map that is specific to update_card, which has PaymentMethod with its respective PaymentMethodType and the connector that it supports ?